src/services/item/schemas.ts

```diff
@@ -0,0 +1,575 @@
+  return Type.Composite(
+    [
+      Type.Omit(itemCreateSchemaFactory(literal, extra), ['settings']),
+      Type.Object({ settings }, { additionalProperties: false }),
+    ],
+    { additionalProperties: false },
+  );
+}
+
+const emptyExtra = Type.Object({}, { additionalProperties: false });
+const fileItemExtra = Type.Object(
+  {
+    name: Type.String(),
+  },
+  { additionalProperties: false },
+);
+const folderItemCreateSchema = itemCreateSchemaFactory(ItemType.FOLDER, emptyExtra);
+const appItemCreateSchema = itemCreateSchemaFactory(
+  ItemType.APP,
+  Type.Object({ url: Type.String({ format: 'uri' }) }, { additionalProperties: false }),
+);
+const documentItemCreateSchema = itemCreateSchemaFactory(
+  ItemType.DOCUMENT,
+  Type.Object(
+    {
+      content: Type.String(),
+      flavor: Type.Optional(Type.Enum(DocumentItemExtraFlavor)),
+      isRaw: Type.Optional(Type.Boolean()),
+    },
+    { additionalProperties: false },
+  ),
+);
+const linkItemCreateSchema = itemCreateSchemaFactoryWithSettings(
+  ItemType.LINK,
+  Type.Object(
+    {
+      url: Type.String({ format: 'uri' }),
+      thumbnails: Type.Optional(Type.Array(Type.String())),
+      icons: Type.Optional(Type.Array(Type.String())),
+      html: Type.Optional(Type.String()),
+      description: Type.Optional(Type.String()),
+    },
+    { additionalProperties: false },
+  ),
+  Type.Composite(
+    [
+      settingsSchema,
+      Type.Object(
+        {
+          showLinkIframe: Type.Optional(Type.Boolean()),
+          showLinkButton: Type.Optional(Type.Boolean()),
+        },
+        { additionalProperties: false },
+      ),
+    ],
+    {
+      additionalProperties: false,
+    },
+  ),
+);
+const localFileItemCreateSchema = itemCreateSchemaFactory(ItemType.LOCAL_FILE, fileItemExtra);
+const s3FileItemCreateSchema = itemCreateSchemaFactory(ItemType.S3_FILE, fileItemExtra);
+const etherpadItemCreateSchema = itemCreateSchemaFactory(
+  ItemType.ETHERPAD,
+  Type.Object({ groupID: Type.String(), padID: Type.String() }, { additionalProperties: false }),
+);
+const h5pItemCreateSchema = itemCreateSchemaFactory(
+  ItemType.H5P,
+  Type.Object(
+    { contentId: Type.String(), h5pFilePath: Type.String(), contentFilePath: Type.String() },
+    { additionalProperties: false },
+  ),
+);
+const shortcutItemCreateSchema = itemCreateSchemaFactory(
+  ItemType.SHORTCUT,
+  Type.Object({ target: customType.UUID() }, { additionalProperties: false }),
+);
+
+export const augmentedItemCreateSchemaRef = registerSchemaAsRef(
+  'augmentedItemCreate',
+  'Augmented Item Create',
+  customType.Discrimineable(
+    [
+      folderItemCreateSchema,
+      appItemCreateSchema,
+      documentItemCreateSchema,
+      linkItemCreateSchema,
+      localFileItemCreateSchema,
+      s3FileItemCreateSchema,
+      etherpadItemCreateSchema,
+      h5pItemCreateSchema,
+      shortcutItemCreateSchema,
+    ],
+    'type',
+  ),
+);
```

Comment:
Maybe it makes sense to move all "createSchema" related code to its own file?
Or maybe we agree on having an endpoint for each type it's not necessary.